mmap+memset+fork.c

```diff
@@ -0,0 +1,145 @@
+#include <stdio.h>
+#include <sys/mman.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <getopt.h>
+
+int flag = MAP_ANONYMOUS | MAP_POPULATE|MAP_FIXED;
+int do_fork = 0;
+int demo = 0;
+int debug_print = 0;
+int i_from = 30;
+int i_till = 39;
+int map_size = 4096;
+int set_size = 4096;
+
+void usage(){
+	printf("Usage: ./mmap+memset+fork [OPTION]\n");	
+	printf("Tests virtual adress space updates with fork and mmap options\n\n");
+	printf("-d, --debug,		prints out the PID and address touched by memset\n");
+	printf("-k, --keystroke,	same as debug, but requires a char to b entered to show each line\n");
+	printf("-h, --help,		prints out help message\n");
+	printf("-s, --map_shared,	sets MAP_SHARED flag in mmap\n");
+	printf("-p, --map_private,	sets MAP_PRIVATE flag in mmap\n");
+	printf("-f, --fork_child,	causes program to fork a child process\n");
+	printf("-g, --from,		mmaps and memsets from 2^(arg)\n");
+	printf("-t, --till,		mmaps and memsets till 2^(arg)\n");
+	printf("-a, --map_size,		mmaps given size\n");
+	printf("-b, --set_size,		memsets given size\n");
```

Comment:
We may need these letters later, but how about `-m` for `--map_size` and `-s` for `--set_size`?

Reply:
nitpick - From the [Program Argument Syntax Conventions](https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html):
> GNU adds long options to these conventions. Long options consist of ‘--’ followed by a name made of alphanumeric characters and dashes. Option names are typically one to three words long, with **hyphens** to separate words.

You choice though, not a deal breaker :)

Reply:
s is already used for map_shared

Reply:
You're right, we're running out of good letters.  I wonder if getopt will let us use `-S` and `-P` (capital letters) for the shared/private options.  I don't have a strong opinion on selecting these, so whatever you think would be best.